src/utils/buildTaggedTemplate.js

```diff
@@ -0,0 +1,88 @@
+import { dirname, relative } from 'path';
+import { stripIndent } from 'common-tags';
+
+function resolveInterpolation(path, nodeMap, localStyle) {
+  let nextPath = path.resolve();
+  while (nextPath && nextPath.isMemberExpression()) {
+    nextPath = nextPath.get('object').resolve();
+  }
+  const style = nextPath && nodeMap.get(nextPath.node);
+  if (!style) return null;
+
+  return {
+    externalName: style.isClassNames
+      ? path.get('property').node.name
+      : style.name,
+    importName: relative(
+      dirname(localStyle.absoluteFilePath),
+      style.absoluteFilePath,
+    ),
+  };
+}
+
+export default (path, nodeMap, localStyle, { tagName, styledTag }) => {
+  const { quasi, tag } = path.node;
+
+  const interpolations = new Map();
+  const expressions = path.get('quasi.expressions');
+
+  let text = '';
+  let id = 1;
+
+  quasi.quasis.forEach((tmplNode, idx) => {
+    const { cooked } = tmplNode.value;
+    const expr = expressions[idx];
+
+    text += cooked;
+
+    if (expr) {
+      const result = expr.evaluate();
+
+      if (result.confident) {
+        text += result.value;
+        return;
+      }
+
+      const isCssTag = tag.name === tagName;
+
+      if (!isCssTag) {
+        const interpolation = resolveInterpolation(expr, nodeMap, localStyle);
+
+        if (interpolation) {
+          const localName = `a${id++}`;
```

Comment:
do we not want to re-use these when dealing with equivalent expressions?

Reply:
hmm good point, I don't think matters ultimately since the compiled version would be the same...there isn't much harm in checking tho